src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -29,42 +30,43 @@ public Range(int first, uint length)
                 throw new ArgumentOutOfRangeException(nameof(first));
             
             First = first;
-            Last = (int)(first + length);
+            End = (int)(first + length);
 
-            if (Last < First) throw new ArgumentOutOfRangeException(nameof(length));
+            if (End < First) throw new ArgumentOutOfRangeException(nameof(length));
         }
 
-        private Range(int first, int last)
+        private Range(int first, int end)
         {
             First = first;
-            Last = last;
+            End = end;
         }
 
-        public bool IsBound => First != UnboundedFirst && Last != UnboundedLast;
+        public bool IsBound => First != UnboundedFirst && End != UnboundedEnd;
```

Comment:
`IsBound` should take into account that the range could be reversed:

* `Range(UnboundedLast, UnboundedFirst)` is unbound;
* `Range(5, UnboundedFirst)` is unbound;
* `Range(UnboundedLast, 5)` is unbound too.